baseapp/test_helpers.go

```diff
@@ -64,11 +65,15 @@ func (app *BaseApp) NewContext(isCheckTx bool) sdk.Context {
 }
 
 func (app *BaseApp) NewUncachedContext(isCheckTx bool, header cmtproto.Header) sdk.Context {
+	cmtHeader, _ := cmttypes.HeaderFromProto(&header)
```

Comment:
I think its fine to discard error here. If you pass a bad header that's your own fault imo.

Calling `cmtHeader.Hash()` on nil will produce `nil` - https://github.com/cometbft/cometbft/blob/main/types/block.go#L447-L449